flows/interfaces.go

```diff
@@ -163,12 +163,17 @@ type Context interface {
 }
 
 type Event interface {
-	CreatedOn() *time.Time
+	CreatedOn() time.Time
 	SetCreatedOn(time.Time)
 
 	Step() StepUUID
 	SetStep(StepUUID)
 
+	FromCaller() bool
+	SetFromCaller(bool)
+
+	Apply(FlowRun)
```

Comment:
Currently the actions catch the error cases and turn them into error events, but yeah seems ok for events to generate error events - certainly if they come from the caller.